src/Umbraco.Core/IO/MediaPathSchemes/TwoGuidsMediaPathScheme.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.IO;
+
+namespace Umbraco.Core.IO.MediaPathSchemes
+{
+    /// <summary>
+    /// Implements a two-guids media path scheme.
+    /// </summary>
+    /// <remarks>
+    /// <para>Path is "{itemGuid}-{propertyGuid}/{filename}".</para>
+    /// </remarks>
+    public class TwoGuidsMediaPathScheme : IMediaPathScheme
+    {
+        /// <inheritdoc />
+        public void Initialize(IFileSystem filesystem)
+        { }
+
+        /// <inheritdoc />
+        public string GetFilePath(Guid itemGuid, Guid propertyGuid, string filename, string previous = null)
+        {
+            var directory = itemGuid.ToString("N") + "-" + propertyGuid.ToString("N");
```

Comment:
I think it would be nicer to have it like Guid/Guid instead of Guid-Guid so that sub folders are created per item. I'll test.